core/src/sumeragi/mod.rs

```diff
@@ -26,6 +25,7 @@ use network_topology::{Role, Topology};
 
 use crate::handler::ThreadHandler;
 
+/// The main event loop that powers sumeragi.
```

Comment:
I see that in other cases module-level doc tests are *usually* located inside of a module, not outside of it.